core/src/wsv.rs

```diff
@@ -215,16 +226,19 @@ impl<W: WorldTrait> WorldStateView<W> {
         self.new_block_notifier.subscribe()
     }
 
-    /// Send events to known subscribers.
-    pub(crate) fn produce_events(&self, events: impl IntoIterator<Item = DataEvent>) {
-        let events = events.into_iter().map(Event::from);
+    /// Send `events` to known subscribers.
+    fn produce_events<Events>(&self, events: Events)
+    where
+        Events: IntoIterator<Item = DataEvent>,
```

Comment:
```suggestion
        E: IntoIterator<Item = DataEvent>,
```
This is an odd convention, but generic parameters are normally given a single letter, to distinguish from specific types.